content/docs/hooks-faq.md

```diff
@@ -4,149 +4,148 @@ title: Hooks FAQ
+### Apakah *Hooks* dapat berjalan dengan *static typing*?{#do-hooks-work-with-static-typing}
 
-Hooks were designed with static typing in mind. Because they're functions, they are easier to type correctly than patterns like higher-order components. The latest Flow and TypeScript React definitions include support for React Hooks.
+Hooks didesain dengan *static typing* pada dasarnya. Karena hal-hal tersebut adalah fungsi, *static typing* lebih mudah ditulis dengan benar daripada pola-pola seperti *higher-order components*. Definisi *Flow* dan *TypeScript React* sudah termasuk *support* untuk React *Hooks* di dalamnya.
 
-Importantly, custom Hooks give you the power to constrain React API if you'd like to type them more strictly in some way. React gives you the primitives, but you can combine them in different ways than what we provide out of the box.
+Penting untuk diingat, *custom Hooks* memberikan Anda kewenangan untuk memaksa *React API* jika Anda ingin menuliskannya dengan lebih ketat dalam beberapa hal. React memberikan Anda *primitives*, tapi Anda bisa menggabungkannya dengan cara-cara lain yang unik selain dari yang kami sediakan untuk Anda.
 
-### How to test components that use Hooks? {#how-to-test-components-that-use-hooks}
+### Bagaimana cara menguji komponen-komponen yang menggunakan Hooks? {#how-to-test-components-that-use-hooks}
 
-From React's point of view, a component using Hooks is just a regular component. If your testing solution doesn't rely on React internals, testing components with Hooks shouldn't be different from how you normally test components.
+Dari sudut pandang React, sebuah komponen yang menggunakan *Hooks* hanyalah sekedar komponen biasa. Jika solusi *testing* Anda tidak bergantung pada fitur-fitur di dalam React, menguji komponen dengan *Hooks* tidak akan berbeda halnya dengan uji komponen yang normalnya Anda lakukan.
 
->Note
+>Catatan
 >
->[Testing Recipes](/docs/testing-recipes.html) include many examples that you can copy and paste.
+>[Testing Recipes](/docs/testing-recipes.html) memuat banyak contoh yang bisa Anda salin dan terapkan.
 
-For example, let's say we have this counter component:
+Sebagai contoh, umpamakan kita memiliki komponen penghitung (*counter*) berikut:
 
 ```js
 function Example() {
   const [count, setCount] = useState(0);
   useEffect(() => {
-    document.title = `You clicked ${count} times`;
+    document.title = `Anda meng-klik sebanyak ${count} kali`;
   });
   return (
     <div>
-      <p>You clicked {count} times</p>
+      <p>Anda meng-klik sebanyak {count} kali</p>
       <button onClick={() => setCount(count + 1)}>
-        Click me
+        Klik sayaS
       </button>
     </div>
   );
 }
 ```
+Kami akan mengujinya dengan React *DOM*. Pastikan bahwa perilakunya dengan yang terjadi di peramban, kami akan mengemas proses *rendering* kode dan memperbaruinya ke dalam [`ReactTestUtils.act()`](/docs/test-utils.html#act) beberapa pemanggilan:
```

Comment:
```suggestion
Kami akan mengujinya dengan React DOM. Pastikan bahwa perilakunya dengan yang terjadi di peramban, kami akan mengemas proses *rendering* kode dan memperbaruinya ke dalam [`ReactTestUtils.act()`](/docs/test-utils.html#act) beberapa pemanggilan:
```

Abbreviations in foreign languages need not be italicised.
Reference: https://twitter.com/ivanlanin/status/898520073224167424

Reply:
>Kami akan mengujinya dengan React DOM. Pastikan bahwa perilakunya dengan yang terjadi di peramban, kami akan mengemas proses *rendering* kode dan memperbaruinya ke dalam [`ReactTestUtils.act()`](/docs/test-utils.html#act) beberapa pemanggilan:

suggested revision, applied